odoo/orm/models.py

```diff
@@ -4234,6 +4236,24 @@ def create(self, vals_list: list[ValuesType]) -> Self:
         self = self.browse()
         self.check_access('create')
 
+        # check access to all user-provided fields
+        if not isinstance(vals_list, list):
+            # reify when an iterator is passed as argument
+            vals_list = list(vals_list)
+        vals_fields = {k for vals in vals_list for k in vals}
+        vals_fields.update(
+            field_name
+            for k in self.env.context
+            if k.startswith('default_')
+            and (field_name := k[8:])
+            and field_name in self._fields
+        )
+        for field_name in vals_fields:
+            try:
+                self._check_field_access(self._fields[field_name], 'write')
+            except KeyError as e:
+                raise ValueError(f"Invalid field {field_name!r} in {self._name!r}") from e
```

Comment:
maybe we could `try` `except` only for `self._fields[field_name]` in case the `KeyError` is raised in the override method of ` _has_field_access`